server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -858,42 +867,53 @@ public final <T extends TransportResponse> void sendRequest(
+                );
+                final TransportResponseHandler<T> delegate;
+                if (request.getParentTask().isSet()) {
+                    // TODO: capture the connection instead so that we can cancel child tasks on the remote connections.
+                    final Releasable unregisterChildNode = taskManager.registerChildNode(
+                        request.getParentTask().getId(),
+                        connection.getNode()
+                    );
+                    delegate = new TransportResponseHandler<T>() {
+                        @Override
+                        public void handleResponse(T response) {
+                            unregisterChildNode.close();
+                            traceableTransportResponseHandler.handleResponse(response);
+                        }
 
-                    @Override
-                    public void handleException(TransportException exp) {
-                        unregisterChildNode.close();
-                        handler.handleException(exp);
-                    }
+                        @Override
+                        public void handleException(TransportException exp) {
+                            unregisterChildNode.close();
+                            traceableTransportResponseHandler.handleException(exp);
+                        }
 
-                    @Override
-                    public String executor() {
-                        return handler.executor();
-                    }
+                        @Override
+                        public String executor() {
+                            return traceableTransportResponseHandler.executor();
+                        }
 
-                    @Override
-                    public T read(StreamInput in) throws IOException {
-                        return handler.read(in);
-                    }
+                        @Override
+                        public T read(StreamInput in) throws IOException {
+                            return traceableTransportResponseHandler.read(in);
+                        }
 
-                    @Override
-                    public String toString() {
-                        return getClass().getName() + "/[" + action + "]:" + handler.toString();
-                    }
-                };
-            } else {
-                delegate = handler;
+                        @Override
+                        public String toString() {
+                            return getClass().getName() + "/[" + action + "]:" + traceableTransportResponseHandler.toString();
+                        }
+                    };
+                } else {
+                    delegate = traceableTransportResponseHandler;
+                }
+                asyncSender.sendRequest(connection, action, request, options, delegate);
             }
-            asyncSender.sendRequest(connection, action, request, options, delegate);
         } catch (final Exception ex) {
```

Comment:
Referring https://github.com/opensearch-project/OpenSearch/pull/9415/files#r1316482812, we assume here that the transport action is invoked and the span will be finished at some point, but we didn't take into account the the `asyncSender.sendRequest` may just fail with an exception earlier before: the span should be created before and the  `traceableTransportResponseHandler.handleException(te);` should be called (instead of `handler.handleException(te);`). 

Please bring back the testing verification, we will work through the issues one by one.